lib/MetaCPAN/Server.pm

```diff
@@ -150,10 +150,13 @@ sub read_param {
 # with a not_found message
 sub stash_or_detach {
     my ( $c, $data ) = @_;
-    $data
-        ? $c->stash($data)
-        : $c->detach( '/not_found',
-        ['The requested info could not be found'] );
+    if ($data) {
+        $c->stash($data);
+    }
+    else {
+        $c->cdn_stale_if_error('1M');    # taken from Root::end()
```

Comment:
I don't see that it does... actually, I don't see any calls to end()... is that being called implicitly by the framework?

I'll add the logic then.